won node crashes with a JMSException

In some settings, the WoN node crashes after some time after a peer (e.g. the DebugBot) is stopped. 

This seems to be connected to an ActiveMQ bug in versions 5.7.0 and 5.8.0: https://issues.apache.org/jira/browse/AMQ-4366

If it is, there is a workaround: setting the connectionFactory's idle timeout to 0. We may have to upgrade our dependency in the long run, though.

Here is a typical log snippet/stack trace:
```
Feb 19, 2018 12:13:23 PM org.apache.catalina.core.StandardContext reload
INFORMATION: Reloading Context with name [/won] has started
Feb 19, 2018 12:13:23 PM org.apache.catalina.core.ApplicationContext log
INFORMATION: Destroying Spring FrameworkServlet 'linkedDataPageServlet'
19.2.18 12:13:23.450 [ContainerBackgroundProcessor[StandardEngine[Catalina]]] INFO  o.s.w.c.s.XmlWebApplicationContext - Closing WebApplicationContext for namespace 'linkedDataPageServlet-servlet': startup date [Mon Feb 19 12:12:53 CET 2018]; parent: Root WebApplicationContext
Feb 19, 2018 12:13:23 PM org.apache.catalina.core.ApplicationContext log
INFORMATION: Closing Spring root WebApplicationContext
19.2.18 12:13:23.474 [ContainerBackgroundProcessor[StandardEngine[Catalina]]] INFO  o.s.w.c.s.XmlWebApplicationContext - Closing Root WebApplicationContext: startup date [Mon Feb 19 12:12:09 CET 2018]; root of context hierarchy
19.2.18 12:13:23.479 [ContainerBackgroundProcessor[StandardEngine[Catalina]]] INFO  o.s.s.c.ThreadPoolTaskScheduler - Shutting down ExecutorService 'taskScheduler'
19.2.18 12:13:23.480 [ContainerBackgroundProcessor[StandardEngine[Catalina]]] INFO  o.a.x.s.c.ResourceXmlApplicationContext - Closing org.apache.xbean.spring.context.ResourceXmlApplicationContext@261b2fd7: startup date [Mon Feb 19 12:12:46 CET 2018]; parent: Root WebApplicationContext
19.2.18 12:13:23.481 [ContainerBackgroundProcessor[StandardEngine[Catalina]]] INFO  o.a.activemq.broker.BrokerService - Apache ActiveMQ 5.8.0 (wonBroker, ID:SAT017-53689-1519038767386-0:1) is shutting down
19.2.18 12:13:24.687 [ContainerBackgroundProcessor[StandardEngine[Catalina]]] INFO  o.a.a.broker.TransportConnector - Connector ssl Stopped
19.2.18 12:13:24.687 [ContainerBackgroundProcessor[StandardEngine[Catalina]]] INFO  o.a.a.broker.TransportConnector - Connector vm Stopped
19.2.18 12:13:24.695 [ContainerBackgroundProcessor[StandardEngine[Catalina]]] INFO  o.a.a.broker.TransportConnector - Connector vm://wonBroker Stopped
19.2.18 12:13:24.716 [ActiveMQ Connection Executor: ssl:///192.168.124.49:61616] WARN  o.s.j.c.CachingConnectionFactory - Encountered a JMSException - resetting the underlying JMS Connection
javax.jms.JMSException: java.io.EOFException
	at org.apache.activemq.util.JMSExceptionSupport.create(JMSExceptionSupport.java:49) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.ActiveMQConnection.onAsyncException(ActiveMQConnection.java:1969) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.ActiveMQConnection.onException(ActiveMQConnection.java:1988) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.transport.TransportFilter.onException(TransportFilter.java:101) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.transport.ResponseCorrelator.onException(ResponseCorrelator.java:126) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.transport.TransportFilter.onException(TransportFilter.java:101) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.transport.TransportFilter.onException(TransportFilter.java:101) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.transport.WireFormatNegotiator.onException(WireFormatNegotiator.java:160) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.transport.AbstractInactivityMonitor.onException(AbstractInactivityMonitor.java:332) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.transport.TransportSupport.onException(TransportSupport.java:96) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.transport.tcp.TcpTransport.run(TcpTransport.java:200) ~[activemq-client-5.8.0.jar:5.8.0]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_151]
Caused by: java.io.EOFException: null
	at java.io.DataInputStream.readInt(DataInputStream.java:392) ~[na:1.8.0_151]
	at org.apache.activemq.openwire.OpenWireFormat.unmarshal(OpenWireFormat.java:275) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.transport.tcp.TcpTransport.readCommand(TcpTransport.java:221) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.transport.tcp.TcpTransport.doRun(TcpTransport.java:213) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.transport.tcp.TcpTransport.run(TcpTransport.java:196) ~[activemq-client-5.8.0.jar:5.8.0]
	... 1 common frames omitted
19.2.18 12:13:24.723 [Camel (wonOwnerCamel) thread #11 - TemporaryQueueReplyManager[OwnerProtocol.in]] WARN  o.a.c.c.j.r.TemporaryQueueReplyManager - Exception inside the DMLC for Temporary ReplyTo Queue for destination OwnerProtocol.in, refreshing ReplyTo destination
javax.jms.IllegalStateException: The Consumer is closed
	at org.apache.activemq.ActiveMQMessageConsumer.checkClosed(ActiveMQMessageConsumer.java:817) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.ActiveMQMessageConsumer.receive(ActiveMQMessageConsumer.java:581) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.springframework.jms.support.destination.JmsDestinationAccessor.receiveFromConsumer(JmsDestinationAccessor.java:130) ~[spring-jms-4.3.8.RELEASE.jar:4.3.8.RELEASE]
	at org.springframework.jms.listener.AbstractPollingMessageListenerContainer.receiveMessage(AbstractPollingMessageListenerContainer.java:416) ~[spring-jms-4.3.8.RELEASE.jar:4.3.8.RELEASE]
	at org.springframework.jms.listener.AbstractPollingMessageListenerContainer.doReceiveAndExecute(AbstractPollingMessageListenerContainer.java:302) ~[spring-jms-4.3.8.RELEASE.jar:4.3.8.RELEASE]
	at org.springframework.jms.listener.AbstractPollingMessageListenerContainer.receiveAndExecute(AbstractPollingMessageListenerContainer.java:255) ~[spring-jms-4.3.8.RELEASE.jar:4.3.8.RELEASE]
	at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.invokeListener(DefaultMessageListenerContainer.java:1166) ~[spring-jms-4.3.8.RELEASE.jar:4.3.8.RELEASE]
	at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.executeOngoingLoop(DefaultMessageListenerContainer.java:1158) ~[spring-jms-4.3.8.RELEASE.jar:4.3.8.RELEASE]
	at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.run(DefaultMessageListenerContainer.java:1055) ~[spring-jms-4.3.8.RELEASE.jar:4.3.8.RELEASE]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_151]
19.2.18 12:13:24.723 [Camel (wonOwnerCamel) thread #11 - TemporaryQueueReplyManager[OwnerProtocol.in]] WARN  o.a.c.c.j.DefaultJmsMessageListenerContainer - Setup of JMS message listener invoker failed for destination 'temporary' - trying to recover. Cause: The Consumer is closed
19.2.18 12:13:24.723 [Camel (wonOwnerCamel) thread #15 - JmsConsumer[OwnerProtocol.Out.427831685025597b857ab154680b9dacd165bfc4d26b4f99b4d215ab]] WARN  o.a.c.c.j.DefaultJmsMessageListenerContainer - Setup of JMS message listener invoker failed for destination 'OwnerProtocol.Out.427831685025597b857ab154680b9dacd165bfc4d26b4f99b4d215ab' - trying to recover. Cause: The Consumer is closed
19.2.18 12:13:24.726 [ContainerBackgroundProcessor[StandardEngine[Catalina]]] INFO  o.a.activemq.broker.BrokerService - Apache ActiveMQ 5.8.0 (wonBroker, ID:SAT017-53689-1519038767386-0:1) uptime 37.451 seconds
19.2.18 12:13:24.727 [ContainerBackgroundProcessor[StandardEngine[Catalina]]] INFO  o.a.activemq.broker.BrokerService - Apache ActiveMQ 5.8.0 (wonBroker, ID:SAT017-53689-1519038767386-0:1) is shutdown
19.2.18 12:13:24.730 [ContainerBackgroundProcessor[StandardEngine[Catalina]]] INFO  o.s.c.e.EhCacheManagerFactoryBean - Shutting down EhCache CacheManager 'nodeCacheManager'
19.2.18 12:13:24.754 [ContainerBackgroundProcessor[StandardEngine[Catalina]]] INFO  o.a.camel.spring.SpringCamelContext - Apache Camel 2.12.1 (CamelContext: wonNodeCamel) is shutting down
19.2.18 12:13:24.757 [ContainerBackgroundProcessor[StandardEngine[Catalina]]] INFO  o.a.c.impl.DefaultShutdownStrategy - Starting to graceful shutdown 21 routes (timeout 300 seconds)
19.2.18 12:13:25.744 [Camel (wonOwnerCamel) thread #15 - JmsConsumer[OwnerProtocol.Out.427831685025597b857ab154680b9dacd165bfc4d26b4f99b4d215ab]] ERROR o.a.c.c.j.DefaultJmsMessageListenerContainer - Could not refresh JMS Connection for destination 'OwnerProtocol.Out.427831685025597b857ab154680b9dacd165bfc4d26b4f99b4d215ab' - retrying using FixedBackOff{interval=5000, currentAttempts=0, maxAttempts=unlimited}. Cause: Could not connect to broker URL: ssl://192.168.124.49:61616. Reason: java.net.ConnectException: Connection refused: connect
19.2.18 12:13:26.759 [Camel (wonOwnerCamel) thread #11 - TemporaryQueueReplyManager[OwnerProtocol.in]] WARN  o.a.c.c.j.r.TemporaryQueueReplyManager - Exception inside the DMLC for Temporary ReplyTo Queue for destination OwnerProtocol.in, refreshing ReplyTo destination
javax.jms.JMSException: Could not connect to broker URL: ssl://192.168.124.49:61616. Reason: java.net.ConnectException: Connection refused: connect
	at org.apache.activemq.util.JMSExceptionSupport.create(JMSExceptionSupport.java:35) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.ActiveMQConnectionFactory.createActiveMQConnection(ActiveMQConnectionFactory.java:293) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.ActiveMQConnectionFactory.createActiveMQConnection(ActiveMQConnectionFactory.java:238) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.ActiveMQConnectionFactory.createConnection(ActiveMQConnectionFactory.java:184) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.springframework.jms.connection.SingleConnectionFactory.doCreateConnection(SingleConnectionFactory.java:365) ~[spring-jms-4.3.8.RELEASE.jar:4.3.8.RELEASE]
	at org.springframework.jms.connection.SingleConnectionFactory.initConnection(SingleConnectionFactory.java:305) ~[spring-jms-4.3.8.RELEASE.jar:4.3.8.RELEASE]
	at org.springframework.jms.connection.SingleConnectionFactory.getConnection(SingleConnectionFactory.java:283) ~[spring-jms-4.3.8.RELEASE.jar:4.3.8.RELEASE]
	at org.springframework.jms.connection.SingleConnectionFactory.createConnection(SingleConnectionFactory.java:224) ~[spring-jms-4.3.8.RELEASE.jar:4.3.8.RELEASE]
	at org.springframework.jms.support.JmsAccessor.createConnection(JmsAccessor.java:180) ~[spring-jms-4.3.8.RELEASE.jar:4.3.8.RELEASE]
	at org.springframework.jms.listener.AbstractJmsListeningContainer.createSharedConnection(AbstractJmsListeningContainer.java:413) ~[spring-jms-4.3.8.RELEASE.jar:4.3.8.RELEASE]
	at org.springframework.jms.listener.AbstractJmsListeningContainer.refreshSharedConnection(AbstractJmsListeningContainer.java:398) ~[spring-jms-4.3.8.RELEASE.jar:4.3.8.RELEASE]
	at org.springframework.jms.listener.DefaultMessageListenerContainer.refreshConnectionUntilSuccessful(DefaultMessageListenerContainer.java:925) [spring-jms-4.3.8.RELEASE.jar:4.3.8.RELEASE]
	at org.springframework.jms.listener.DefaultMessageListenerContainer.recoverAfterListenerSetupFailure(DefaultMessageListenerContainer.java:899) [spring-jms-4.3.8.RELEASE.jar:4.3.8.RELEASE]
	at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.run(DefaultMessageListenerContainer.java:1077) [spring-jms-4.3.8.RELEASE.jar:4.3.8.RELEASE]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_151]
Caused by: java.net.ConnectException: Connection refused: connect
	at java.net.DualStackPlainSocketImpl.waitForConnect(Native Method) ~[na:1.8.0_151]
	at java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:85) ~[na:1.8.0_151]
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350) ~[na:1.8.0_151]
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206) ~[na:1.8.0_151]
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188) ~[na:1.8.0_151]
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172) ~[na:1.8.0_151]
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392) ~[na:1.8.0_151]
	at java.net.Socket.connect(Socket.java:589) ~[na:1.8.0_151]
	at sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:673) ~[na:1.8.0_151]
	at org.apache.activemq.transport.tcp.TcpTransport.connect(TcpTransport.java:496) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.transport.tcp.TcpTransport.doStart(TcpTransport.java:459) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.util.ServiceSupport.start(ServiceSupport.java:55) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.transport.AbstractInactivityMonitor.start(AbstractInactivityMonitor.java:140) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.transport.TransportFilter.start(TransportFilter.java:58) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.transport.WireFormatNegotiator.start(WireFormatNegotiator.java:72) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.transport.TransportFilter.start(TransportFilter.java:58) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.transport.TransportFilter.start(TransportFilter.java:58) ~[activemq-client-5.8.0.jar:5.8.0]
	at org.apache.activemq.ActiveMQConnectionFactory.createActiveMQConnection(ActiveMQConnectionFactory.java:273) ~[activemq-client-5.8.0.jar:5.8.0]
	... 13 common frames omitted
19.2.18 12:13:26.759 [Camel (wonOwnerCamel) thread #11 - TemporaryQueueReplyManager[OwnerProtocol.in]] ERROR o.a.c.c.j.DefaultJmsMessageListenerContainer - Could not refresh JMS Connection for destination 'temporary' - retrying using FixedBackOff{interval=5000, currentAttempts=0, maxAttempts=unlimited}. Cause: Could not connect to broker URL: ssl://192.168.124.49:61616. Reason: java.net.ConnectException: Connection refused: connect
```

--- a/webofneeds/pom.xml
+++ b/webofneeds/pom.xml
@@ -908,7 +908,7 @@
         <de.uni_koblenz.aggrimm.icp.version>1.0.1</de.uni_koblenz.aggrimm.icp.version>
         <camel.version>2.12.1</camel.version>
         <qpid.version>0.24</qpid.version>
-        <activemq.version>5.8.0</activemq.version>
+        <activemq.version>5.10.2</activemq.version>
         <activemq.conf>classpath:</activemq.conf>
         <jms.version>1.1</jms.version>
         <xbean.version>3.14</xbean.version>   <!-- Check later if it is really needed-->

